src/Impostor.Api/Innersloth/GameOptions/GameOptionsFactory.cs

```diff
@@ -73,4 +74,60 @@ public static void DeserializeInto(IMessageReader reader, IGameOptions gameOptio
                 throw new ArgumentOutOfRangeException(nameof(gameMode), gameMode, null);
         }
     }
+
+    public static byte[] ToBytes(IGameOptions gameOptions, bool forceAprilFoolsMode)
+    {
+        var version = gameOptions.Version;
+
+        if (version < 7)
+        {
+            throw new NotImplementedException("Can not serialize legecy game options");
+        }
+
+        var gameModes = gameOptions.GameMode;
+        if (forceAprilFoolsMode)
+        {
+            switch (gameModes)
+            {
+                case GameModes.Normal:
+                case GameModes.NormalFools:
+                    gameModes = GameModes.NormalFools;
+                    break;
+                case GameModes.HideNSeek:
+                case GameModes.SeekFools:
+                    gameModes = GameModes.SeekFools;
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException("Gamemode does not exist.");
+            }
+        }
+
+        using var writer = MessageWriter.Get(MessageType.Unreliable);
+        writer.Write(version);
+        writer.StartMessage(0);
```

Comment:
FYI, you need to serialize the GameMode first otherwise it fails to parse. Fixed it in my version